source/particle/world.cc

```diff
@@ -731,14 +731,16 @@ namespace aspect
       do
         {
           advect_particles();
+          if (property_manager->need_update() == Property::update_time_substep)
+            update_particles();
         }
       // Keep calling the integrator until it indicates it is finished
       while (integrator->new_integration_step());
 
       apply_particle_per_cell_bounds();
 
       // Update particle properties
-      if (property_manager->need_update() == Property::update_time_step)
+      if (property_manager->need_update() == Property::update_time_step || property_manager->need_update() == Property::update_time_substep)
```

Comment:
Do you need the `||` here? You already called `update_particles` in line 735 above, and nothing has changed since then (a few particles may have been removed or new added in line 740, but that is unrelated).

Reply:
Yes, I think you're actually calling it twice.

Reply:
I modelled it after how the integrators work internally: call supbsteps and then have a seperate call to combine them, but I now realize that that would be quite expensive for because a update is just expensive. Since it is not needed, I will remove this line change.